### PR TITLE
fix(libpng): fix memory leak when draw buf alloc failed

### PR DESCRIPTION
**Pull request by lion2tomato**
fix(libpng): fix memory leak when draw buf alloc failed
fix(imagecache): add cache evict when cache memory is not enough

### CODE REVIEW

File: src/drivers/nuttx/lv_nuttx_image_cache.c [hunk 1]
```diff
@@ -60,7 +64,7 @@ static void free_cb(void * draw_buf);
  *   GLOBAL FUNCTIONS
  **********************/
 
-void lv_nuttx_image_cache_init(void)
+void lv_nuttx_image_cache_init(bool use_independent_image_heap)
```
Comment (FASTSHIFT) [inline src/drivers/nuttx/lv_nuttx_image_cache.c:67]: Please send a separate PR for nuttx image cache modification.

**Review by FASTSHIFT — CHANGES_REQUESTED** [review]

File: src/drivers/nuttx/lv_nuttx_image_cache.h [hunk 2]
```diff
@@ -15,6 +15,7 @@ extern "C" {
  *********************/
 
 #include "../../lv_conf_internal.h"
+#include <stdbool.h>
```
Comment (kisvegabor) [inline src/drivers/nuttx/lv_nuttx_image_cache.h:18]:
```suggestion
#include LV_STDBOOL_INCLUDE
```
